frame/bridge/relayer-game/src/lib.rs

```diff
@@ -88,93 +94,85 @@ impl WeightInfo for () {}
-		/// Pending header rejected. [block number]
-		PendingHeaderRejected(TcBlockNumber),
+		TODO(GameId),
 	}
 }
 
 decl_error! {
 	pub enum Error for Module<T: Trait<I>, I: Instance> {
-		/// Active Game - TOO MANY
-		ActiveGameTM,
-
-		/// Can not bond with value less than usable balance.
-		InsufficientBond,
-
-		/// Pending Header - NOT FOUND
-		PendingHeaderNF,
-
-		/// Proposal - INVALID
-		ProposalI,
-
-		/// Proposal - ALREADY EXISTED
-		ProposalAE,
-
+		/// Relay Stuffs - ALREADY CONFIRMED
+		RelayStuffsAC,
 		/// Round - MISMATCHED
 		RoundMis,
-
-		/// Target Header - ALREADY CONFIRMED
-		TargetHeaderAC,
+		/// Active Games - TOO MANY
+		ActiveGamesTM,
+		/// Game - CLOSED
+		GameC,
+		/// Proposal - DUPLICATED
+		ProposalDup,
+		/// Usable *RING* for Bond - INSUFFICIENT
+		BondIns,
+		/// Proofses Quantity - INVALID
+		ProofsesInv,
+		/// Proposal - NOT EXISTED
+		ProposalNE,
+		/// Extended Proposal - NOT EXISTED
+		ExtendedProposalNE,
+		/// Previous Proofs - INCOMPLETE
+		PreviousProofsInc,
 	}
 }
 
 decl_storage! {
 	trait Store for Module<T: Trait<I>, I: Instance = DefaultInstance> as DarwiniaRelayerGame {
-		/// All the proposals here per game
-		pub Proposals
-			get(fn proposals_of_game)
-			: map hasher(blake2_128_concat) GameId<TcBlockNumber<T, I>>
-			=> Vec<RelayProposalT<T, I>>;
+		/// The number of active games
+		pub ActiveGames get(fn active_games): u8;
 
-		/// All the proposal relay headers(not brief) here per game
-		pub Headers
-			get(fn header_of_game_with_hash)
+		/// All the active games' proposals here
+		///
+		/// The first key is game id, the second key is round index
+		/// then you will get the proposals under that round in that game
+		pub Proposals
+			get(fn proposals_of_game_at)
 			: double_map
-				hasher(blake2_128_concat) GameId<TcBlockNumber<T, I>>,
-				hasher(blake2_128_concat) TcHeaderHash<T, I>
-			=>  TcHeaderThing<T, I>;
+				hasher(identity) GameId<T, I>,
+				hasher(identity) u32
+			=> Vec<RelayProposalT<T, I>>;
 
-		/// The last confirmed block number record of a game when it start
+		/// The last confirmed block id record of a game when it start
 		pub LastConfirmeds
-			get(fn last_confirmed_of_game)
-			: map hasher(blake2_128_concat) GameId<TcBlockNumber<T, I>>
-			=> TcBlockNumber<T, I>;
-
-		/// The allow samples for each game
-		pub Samples
-			get(fn samples_of_game)
-			: map hasher(blake2_128_concat) TcBlockNumber<T, I>
-			=> Vec<Vec<TcBlockNumber<T, I>>>;
-
-		/// The closed rounds which had passed the challenge time at this moment
+			get(fn last_confirmed_block_id_of)
+			: map hasher(identity) GameId<T, I>
+			=> RelayBlockId<T, I>;
+
+		pub RoundCounts
+			get(fn round_count_of)
+			: map hasher(identity) GameId<T, I>
+			=> u32;
+
+		/// All the closed games here
+		///
+		/// Closed games won't accept any proposal
+		pub ClosedGames
+			get(fn game_closed_at_of)
+			: map hasher(identity) GameId<T, I>
+			=> BlockNumber<T>;
```

Comment:
What's the meaning of value, the type of which is ```BlockNumber<T>```

Reply:
Game close at this moment, won't accept any propsal